app/src/main/java/ch/epfl/sweng/hostme/wallet/ExpirationDatePicker.java

```diff
@@ -166,57 +169,85 @@ private void createNotificationChannel() {
+        ArrayList<Long> scheduledTime = new ArrayList<Long>();
+        ArrayList<String> scheduledMessage = new ArrayList<String>();
+        modifyScheduledArray(date, scheduledTime, scheduledMessage);
+
+        int all_reminders_size = RemindersDate.values().length;
+        for (int i = 0; i < scheduledTime.size(); ++i) {
+
+            Intent intent = new Intent(context, ReminderNotification.class);
+            String title = "Document expiration reminder";
+            String message = "Your " + document.getDocumentName().toLowerCase() + " will expire " + scheduledMessage.get(i);
+
+            intent.putExtra(notification_ID_Extra, document.ordinal() * all_reminders_size + i); //same document notifications goes to a different notification
 
             intent.putExtra(titleExtra, title);
             intent.putExtra(messageExtra, message);
 
 
             PendingIntent pend = PendingIntent.getBroadcast(
                     context.getApplicationContext(),
-                    PENDING_ID + i, //PendingIntent a un requestCode different sinon ça override la même PendingIntent.
+                    document.ordinal() * all_reminders_size + i, //PendingIntent a un requestCode different sinon ça override la même PendingIntent.
                     intent,
                     PendingIntent.FLAG_IMMUTABLE | PendingIntent.FLAG_UPDATE_CURRENT
             );
 
             alarmManager.setExactAndAllowWhileIdle(
                     AlarmManager.RTC_WAKEUP,
-                    scheduleTime.get(i),
+                    scheduledTime.get(i),
                     pend
             );
         }
     }
 
-    private ArrayList<Long> getTime(int dayOfMonth, int month, int year) {
+    private void modifyScheduledArray(Calendar targetDate, ArrayList<Long> scheduledTime, ArrayList<String> scheduledMessage) {
 
-        ArrayList<Long> scheduleTime = new ArrayList<Long>();
+        for (RemindersDate period : RemindersDate.values()) {
+            Calendar targetDate_copy = (Calendar) targetDate.clone();
+            targetDate_copy.add(period.timeUnit, period.number);
 
-        Calendar now = Calendar.getInstance();
-        now.add(Calendar.MINUTE, 1);
-        scheduleTime.add(now.getTimeInMillis());
-
-        Calendar now2 = Calendar.getInstance();
-        now2.add(Calendar.MINUTE, 2);
-        scheduleTime.add(now2.getTimeInMillis());
+            Calendar now = Calendar.getInstance();
+            if (targetDate_copy.getTimeInMillis() - now.getTimeInMillis() > 0) {
+                scheduledTime.add(targetDate_copy.getTimeInMillis());
+                scheduledMessage.add(period.message);
+            }
 
-        return scheduleTime;
+        }
 
+        Calendar nowTestDemo = Calendar.getInstance();
+        nowTestDemo.add(Calendar.SECOND, 20);
+        scheduledTime.add(nowTestDemo.getTimeInMillis());
+        scheduledMessage.add("in 2 days");

```

Comment:
This looks like some test code that made it into the final commit. Maybe remove it if it's not needed.

Reply:
Yeah it is for the demo but in the final version we will remove that!